src/python/artm_vs2012.sln

```diff
@@ -1,62 +0,0 @@
-﻿
```

Comment:
I removed it because I have not VS installation and cannot correctly change those files according to new code structure. I suggest you to create those files and add them to the repository again, if you need them.

Nevertheless, I hate any kind of IDE files in public repositories. Not just me, see [StackOverflow explanation](http://stackoverflow.com/questions/14558450/why-ide-specific-project-files-are-ignored-in-git).

> I prefer VS + pylint and dislike PyCharm.

You prefer, but I cant use VS physically (I have not PC with Windows on board), what do you propose should I do? 

Many opensource projects add IDE files to `.gitignore`. See [sklearn](https://github.com/scikit-learn/scikit-learn/blob/master/.gitignore), [gensim](https://github.com/piskvorky/gensim/blob/develop/.gitignore).

Btw, we use CMake. Do you know the main destiny of it? CMake allows each developer setup its own environment and create config files for any IDE, see [wikipedia](https://en.wikipedia.org/wiki/CMake).